rotkehlchen/serialization/deserialize.py

```diff
@@ -450,6 +451,38 @@ def deserialize_optional(input_val: X | None, fn: Callable[[X], Y]) -> Y | None:
     return fn(input_val)
 
 
+def _get_transaction_receipt(
+        tx_hash: EVMTxHash,
+        raw_receipt_data: dict[str, Any] | None,
+        chain_id: ChainID,
+        timestamp: Timestamp,
+        evm_inquirer: 'EvmNodeInquirer',
+) -> dict[str, Any]:
+    """Get the transaction receipt for a tx during deserialization.
+    Handles a special case for Optimism where the official mainnet node should be tried first
+    no matter the default call order, since other nodes may have null L1 fee values.
+    """
+    if raw_receipt_data is not None:
+        return raw_receipt_data
```

Comment:
isn't this a check that you should have before? No need to perform the call if you already have the receipt